### DISCUSSION TITLE
Deposit Bumpkins 🧑‍🌾

### DISCUSSION THREAD

**Discussion by adamhannigan** [sunflower-land/sunflower-land#2591] [Ideas]
The purpose of this discussion is to explore the concept of depositing Bumpkins into the game. 

Currently when playing Sunflower Land your Bumpkin is read from your personal MetaMask wallet and can be used instantly inside of the game. During the creation of the Bumpkin project there were strong reasons why we thought this would be a good idea - over time we've come to realise this is becoming more of a problem than a benefit to the game.

## Why did we create Bumpkins as a separate project?

When creating the Bumpkin NFT project there were a few compelling hypothesises that led to us launching it as a separate project.

1. Players could use Bumpkin NFTs in their personal wallet as Profile Pictures (PFP) on Web3 supported social media sites
2. The project could be standalone from Sunflower Land and encourage non-players and collectors to get involved.
3. We could build Bumpkin specific functionality at Bumpkins.io quickly without breaking our core game.

**Mistakes**

Over the past 6 months we have learnt a lot and discovered we made some mistakes in these assumptions. Some noteworthy observations include:

1. We had little uptake of the PFPs and social media sites are deprecating support for PFP NFTs (e.g. Instagram)
2. The core audience for Bumpkins did not expand to non-players/general collectors
3. Forcing players to an external website (bumpkins.io) to buy + equip was confusing and caused a lot of friction for our players.

Additional issues we discovered in our current architecture include:

1. Maintaining an additional auction system at Bumpkins.io, which commonly had bugs
2. Players needing to withdraw SFL from the game first to mint an item = high friction
3. Players abusing swapping Bumpkins whilst playing the game
4. Equipping items takes a long time and prevents players from quickly changing clothes
5. When a player sells a farm, a Bumpkin is not sold with it. This makes it extremely annoying for buyers

## Proposal

We are proposing that we require players to deposit their Bumpkins + Wearables into the game before they begin using them. This will work exactly the same way other items in Sunflower Land work such as collectibles + resources.

A player will then need to withdraw their Bumpkins + Wearables if they wish to trade with friends or sell to other players on secondary Marketplaces.

**Benefits**

There are numerous benefits to this approach, namely including:

1. We can perform drops inside of Sunflower Land, reusing our current auctioneers + drop system.
2. Players can use SFL + resources on their farm wallet without needing to withdraw and pay the tax.
3. More Sunflower Land resources can be used to craft items - e.g. Crops, minerals, resources
4. We can allow players to instantly craft certain basic wearables (i.e. Red Farmer Shirt) without a transaction. Similar to other Sunflower Land items, they will be minted when a player stores progress on chain.
5. Players can instantly equip clothing inside of the game without a transaction. This will be crucial with the upcoming MMORPG where players can change clothing easier to show it off
6. It is much more difficult for players to abuse swapping multiple Bumpkins, since they need to explicitly deposit/withdraw into our game.
7. Our architecture gets a lot easier to build (i.e. quicker to launch and scale multiple Bumpkins)

## Workflow

To support this proposal, the following work would need to take place.

1. Allow players to Deposit Bumpkins + Wearables onto their farm
2. Allow players to Withdraw Bumpkins + Wearables to their personal wallet
3. Support minting Bumpkin Wearables at our in-game auction

## Conclusion

Thank you for taking the time to read this post. Over time it has become more clear that this is a much needed change which will help the Bumpkins + Sunflower Land go to the next level 🚀 

Any feedback would be much appreciated! 🤗

**Comment by eliasSFL** [comment]
So essentially now we deposit our bumpkin wearables into the farm contract address, and we can freely equip and unequip without needing a transaction? What about the wearables that are currently equipped in the bumpkin contract, will those be transferred to the farm contract?

**Reply by adamhannigan** [reply]
@elias7905 Yes, that is right.

The exact details of the deposit process are yet to solved. Ideally if you have wearables equipped, at the time of deposit they will be taken off the Bumpkin Wallet and sent to the farm wallet 👍 

The plan would be to make it as seamless as possible for a user.

**Reply by eliasSFL** [reply]
@adamhannigan maybe like using the in game deposit system, that contract will read any wearables in the bumpkin contract and transfer those wearables into your farm contract?

**Reply by adamhannigan** [reply]
@elias7905 Yes 😄

**Comment by mGranTY** [comment]
LGTM 🚀

**Comment by vitt0** [comment]
Sounds like a good change to do Imo, I assume that with this change bumpkins wearables are going to have the same type of restrictions to withdraw them from the farm? Like less say X player is growing Kale, until it finish they aren't going to be able to withdraw for example their Green Amulet?

Probably something that you guys still need to decide, but what its going to happend with all the Bumpkin Analytics stats? Mostly curious if they are going to be sync on chain when we use the store progress option or if they are going to have a separed option? And if they are going to be any major changes to the getBumpkins or the visit API to consult this info

**Reply by adamhannigan** [reply]
Yes, we will be able to have a better control over restrictions (including when it can be withdrawn).

In terms of Bumpkin Analytics, due to high gas prices, these are no longer stored on chain. We will be focussing our efforts on improving the public API to grab the Bumpkin data 😄

**Comment by brian-man** [comment]
> Players can instantly equip clothing inside of the game without a transaction.

Been wanting this for months.

**Reply by adamhannigan** [reply]
100%

**Comment by Polysemouse** [comment]
LGTM! I am highly in favor of this proposal.  It solves a lot of isuses with the current bumpkin system.

**Comment by TheDcol** [comment]
Yes yes and yes. Posting my previous thoughts on it:

![Screenshot_20230527-073742](https://github.com/sunflower-land/sunflower-land/assets/108110065/8d5d7118-0896-4eeb-b9c0-6374b1b2a201)

**Comment by NotKeen** [comment]
I was a bit doubtful from the beginning about splitting Bumpkins off, but thought maybe I'm just not open-minded enough haha. Much respect to you and the team tho to recognize the wrong path (I agree) and pivot, even tho much work and hope was invested. The right decisions are often not easy to make haha.

**Comment by adamhannigan** [comment]
Looks like overwhelming support for this. I've worked on a prototype over the weekend to support this infrastructure - will try push into the game as soon as possible 🙏 Thanks!

<img width="493" alt="Screen Shot 2023-05-29 at 12 34 19 pm" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/a458b62d-1ba4-4b58-89cc-fc24535ca5d3">

**Reply by eliasSFL** [reply]
Looking good! Looking forward to the implementation of this

**Comment by adamhannigan** [comment]
Screenshots + Progress Updates. Hoping to launch next week

https://github.com/sunflower-land/sunflower-land/pull/2599

Will need to give players some time to get ready 👍

**Reply by vitt0** [reply]
It's looking great 🚀  

A question about the "skin potions" they are going to become just like another normal wearable or the idea is continue having them as "unique use"? I'm just asking coz I saw them as withdrawables items

**Comment by ShiningMinions** [comment]
TBH I'm surprised bumpkin+farm integration wasn't like this to begin with - when bumpkins got released I thought it would work like this (the same way items are sent/withdrawn from your farm).

I think in the distant future, I would like to see a system where locking your assets into the game is more decentralized, but for now this definitely makes more sense to have it this way - I don't mind doing it :+1: . In fact, I'm more willing to lock-in my bumpkin VS other farm assets.
